src/System.Private.CoreLib/src/System/Threading/ThreadStore.cs

```diff
@@ -0,0 +1,57 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime;
+
+namespace System.Threading
+{
+    internal static class ThreadStore
+    {
+        private static readonly ManualResetEvent _allDone = new ManualResetEvent(false);
+
+        // main thread started by runtime and we have no chance (?) to increment it elsewhere
+        private static int _foregroundRunningCount = 1;
```

Comment:
I guess we don't need it to be `volatile` since we reading/updating it only via `Interlocked/Volatile` static methods.